### ISSUE TITLE
onPrizeDefined trigger error

### ISSUE THREAD

**Issue by hoching87** [IvanAdmaers/react-roulette-pro#2]
![Screenshot 2022-01-24 150705](https://user-images.githubusercontent.com/79739168/150737181-779d2540-a0fc-4fb6-8b85-ff4333d58163.png)

The onPrizeDefined  props trigger multiple times

**Comment by IvanAdmaers** [comment]
Hello, @hoching87! Thank you for your interest in this package.

**TL;DR**
Use useCallback for `onPrizeDefined` handler.
```javascript
  const prizeDefinedHandler = useCallback(() => {
    console.log('Prize defined');
  }, []);
```

**Details**

The reason is when every new spin the `onPrizeDefined` prop resets because that's how React Hooks work. The passed function `onPrizeDefined` as a prop is a new every time.

Take a look at the useEffect that includes a `tansitionend` event listener.
_/src/components/RoulettePro/RoulettePro.js_
```javascript
  useEffect(() => {
    if (!contentRef.current) return;

    const content = contentRef.current;

    const listener = content.addEventListener(
      'transitionend',
      handlePrizeDefined,
    );

    return () => content.removeEventListener('transitionend', listener);
  }, [contentRef, log, handlePrizeDefined]);
```

When the handlePrizeDefined function is:
```javascript
  const handlePrizeDefined = useCallback(() => {
    log('Prize defined 🎁');

    onPrizeDefined();
  }, [onPrizeDefined, log]);
```

If remove the `onPrizeDefined` useEffect dependency the onPrizeDefined will trigger only once. But it's not a good idea to ignore useEffect dependencies.

The solution is to move your `onPrizeDefined` handler to the React useCallback hook.

```javascript
  const prizeDefinedHandler = useCallback(() => {
    console.log('Prize defined');
  }, []);
```

If you have any questions fell free to ask them. Cheers!

**Comment by hoching87** [comment]
Hi @IvanAdmaers , 

Thanks for fast reply and your effort for explaining to me. Very helpful and I learnt something new today. Also, I like your package a lot, very cool.

Good luck on your future work. Cheers!

**Comment by IvanAdmaers** [comment]
I just updated the "Advanced Usage" example ([/example/src/AdvancedUsage.js](https://github.com/IvanAdmaers/react-roulette-pro/blob/main/example/src/AdvancedUsage.js)). Now it's more clearly and easy to understand.

Thank you for your support. All the best to you!
